05/app.js

```diff
@@ -0,0 +1,64 @@
+function Student (saveFirstName, saveLastName) {
+        this.firstName = saveFirstName;
+        this.lastName = saveLastName;
+        this.grades = {}
+}
+
+Student.prototype.addGrade = function(grade,subject) {
+    if(typeof this.grades[subject] === 'undefined') {
+        this.grades[subject] = [];
+    }
+    this.grades[subject].push(grade)
+}
+
+
+Student.prototype.getAverage = function(subject) {
+    
+    if (typeof subject !== 'undefined' ) {
+        if(typeof this.grades[subject] === 'undefined') {
+            return 'Nie ma takiego przedmiotu!'
+        }
+        const mean = getMeanOfArray(this.grades[subject])
+    
+        return mean;
+    }
+
+    if(typeof this.grades === 'undefined') {
+        return 'Nie ma ocen z żadnego przedmiotu!'
+    }
```

Comment:
Raczej to sprawdzenie jest nie potrzebne - na początku przypisujesz do `this.grades` pusty obiekt więc ewentualnie będzie to obiekt :)